lib/fastlane/plugin/wpmreleasetoolkit/actions/common/check_translation_progress.rb

```diff
@@ -0,0 +1,135 @@
+
+        check_results(
+          under_threshold_langs: under_threshold_langs, 
+          threshold: params[:min_acceptable_translation_percentage], 
+          skip_confirm: params[:skip_confirm]
+        ) unless under_threshold_langs.length == 0
+
+        UI.message('Done')
+      end
+
+      # Check the status of the translations and returns the list of violations
+      # 
+      # @param [String] URL to the GlotPress project.
+      # @param [String] The list of codes (in GlotPress format) of the languages to check.   
+      # @param [Bool] Whether to abort on the first found violation or not.
+      # @param [Integer] The minimum acceptable percentage of translated strings.
+      #
+      # @return [Array] The list of violations (array of hashes of "language code" and "current percentage")
+      #
+      def self.check_translations(glotpress_url:, language_codes:, abort_on_violations:, threshold:)
+        under_threshold_langs = []
+
+        data = Fastlane::Helper::GlotPressHelper.get_translation_status_data(glotpress_url: glotpress_url) rescue nil
+        UI.abort_with_message!("Can't retrieve data from #{glotpress_url}") if data.nil? || data.length == 0
+
+        language_codes.each do | language_code |
+          progress = Fastlane::Helper::GlotPressHelper.get_translation_status(
+                                      data: data,
+                                      language_code: language_code) rescue -1
+          
+          if (abort_on_violations)
+            UI.abort_with_message!("Can't get data for language #{language_code}") if (progress == -1) 
+            UI.abort_with_message!("#{language_code} is translated #{progress}% which is under the required #{threshold}%.") if (progress < threshold) 
+          end
+          
+          under_threshold_langs.push({:lang => language_code, :progress => progress }) if (progress < threshold)
+        end
+
+        under_threshold_langs
+      end
+
+      # Report the status of the translations and verify whether to abort or not
+      # 
+      # @param [Array] The list of violations (array of hashes of "language code" and "current percentage")
+      # @param [Integer] The minimum acceptable percentage of translated strings.  
+      # @param [Bool] If true, continue after the report without asking the user. 
+      #
+      def self.check_results(under_threshold_langs:, threshold:, skip_confirm:)
+        message = "The translations for the following languages are below the #{threshold}% threshold:\n"
+        
+        under_threshold_langs.each do | lang |
+          message << " - #{lang[:lang]} is at #{lang[:progress]}%.\n"
+        end
+
+        if skip_confirm 
+          UI.important(message) 
+        else
+          if UI.interactive? 
+            UI.abort_with_message!("Aborted by user!") unless UI.confirm("#{message}Do you want to continue?") 
+          else 
+            UI.abort_with_message!(message)
+          end
```

Comment:
This is not supposed to be run on CI and with the current options there is some control. As a first implementation, I think it's not worth it to add complexity to handle cases which have low caches to happen and so the action defaults to failing if there are warnings when run on CI.